Print wallet info on the page

timothymcmackin opened the pull request:
I'd like to be able to print the wallet info on the page for this example, so I looked up how to do this with Vue Suspense, as in these references:

- https://vuejs.org/guide/built-ins/suspense.html
- https://www.trpkovski.com/2021/09/25/suspense-feature-in-vue-3-with-sfc-script-setup/#use-suspense-component

I use this template:

```
<template>
  <Suspense>
    <template #fallback>
      Loading...
    </template>
    <template #default>
      <div>
        {{ address }}
      </div>
    </template>
  </Suspense>
</template>
```

But I get this error:

```
 [Vue warn]: Component <Anonymous>: setup function returned a promise, but no <Suspense> boundary was found in the parent component tree. A component with async setup() must be nested in a <Suspense> in order to be rendered. 
```

timothymcmackin committed:
Trying to print wallet info on the page

napalmpapalam commented on wallet-access/src/App.vue:
```diff
@@ -1,21 +1,27 @@
+console.log('providerType:', provider.providerType)
 
-getMetamaskWalletInfo()
+const address = provider.address
```
```ts
import { ref } from 'vue'

const PROVIDER_EVENTS: Array<keyof IProvider> = [
  'onInitiated',
  'onConnect',
  'onAccountChanged',
]

const address = ref('')

 const providerEvent = provider[event] as (
        cb: (payload: ProviderEventPayload) => void,
) => void

providerEvent?.call(provider, payload => {
   address.value ||= payload?.address ?? ''
})
```

try this. PS. `const address = provider.address` is not reactive - u write value on creating component lifehook, u need to subscribe to the provider event to update value

timothymcmackin replied:
Looks like we need to export `ProviderEventPayload` from the SDK for this to work.

napalmpapalam replied:
U can use our react implementations as reference

https://github.com/rarimo/js-sdk/blob/main/packages/react-nft-checkout/src/hooks/useProvider.ts

napalmpapalam commented on wallet-access/src/App.vue:
```diff
@@ -1,21 +1,27 @@
+console.log('chainType:', ChainTypes[provider.chainType!])
+console.log('providerType:', provider.providerType)
 
-getMetamaskWalletInfo()
```
bring back this, and remove suspense, it's experimental feature which not recommended to use in the production

timothymcmackin committed:
Revert "Trying to print wallet info on the page"

This reverts commit 3420ef2ba0066203202e8372837c7681852c5d1a.

timothymcmackin committed:
Trying Semen's way of using events

napalmpapalam commented on wallet-access/src/App.vue:
```diff
@@ -10,12 +19,21 @@ const getMetamaskWalletInfo = async () => {
   console.log('Address:', provider.address)
   console.log('chainType:', ChainTypes[provider.chainType!])
   console.log('providerType:', provider.providerType)
+
```
`address.value = provider?.address ?? ''`

timothymcmackin committed:
This is not available from the SDK yet

timothymcmackin committed:
Add this assignment

timothymcmackin commented:
@napalmpapalam Ok, I added that statement and removed the unavailable ProviderEventPayload type and it seems to work if I am already logged in to metamask before I open the page. If I have to log in to the wallet, the page does not show my address. Maybe something to do with the events?

The `PROVIDER_EVENTS` object and the `providerEvent?.call()` callback do not seem to be used. Are you using these for some forward-thinking reason?

timothymcmackin committed:
Use the react scenario as a guide for handling these events
